server/middlewares/auth/index.js

```diff
@@ -0,0 +1,20 @@
+const Boom = require('@hapi/boom');
+const { verifyToken } = require('../../utils');
+
+const Auth = async (req, res, next) => {
+  const { token } = req.cookies;
+  if (token) {
+    try {
+      req.user = await verifyToken(token);
+      next();
+    } catch (err) {
+      throw Boom.unauthorized('You are not Authorized');
+    }
+  } else {
+    res
+      .status(401)
+      .json({ statusCode: 401, message: 'You are not registered yet' });
+  }
```

Comment:
It might be simpler if you handled "not registered yet" and "not authorized" in the same way (have them both pass an error to `next`, or both set `res.status` and `res.json`), instead of handling the two errors in two different ways.  Let me know if that doesn't make sense and I can explain in more detail.

Reply:
yes, please explain in detail

Reply:
Sure
Right now, you throw an error for invalid tokens:
```
try {
      req.user = await verifyToken(token);
      next();
    } catch (err) {
      throw Boom.unauthorized('You are not Authorized');
  }
```

But if the token is missing, you do something a little differently - you set the status on `res`:
```
res
      .status(401)
      .json({ statusCode: 401, message: 'You are not registered yet' });
```

I think it would make the code a little simpler (and therefore easier to change or understand, for others in the future) if you made the two errors use the same style of error handling.  For example, maybe they could both throw:
```
const { token } = req.cookies;
if (!token) throw Boom.unauthorized('You are not registered yet');
try {
  req.user = await verifyToken(token);
  next();
} catch (err) {
  throw Boom.unauthorized('You are not Authorized');
}
```
Also, you could change the both to `next(err)` which I think was suggested earlier.

Reply:
I'll change it, Thanks Tom